Presentation.Web/app/services/exportGridToExcelService.ts

```diff
@@ -1,66 +1,92 @@
 ﻿module Kitos.Services.System {
+    import ArrayHelper = Helpers.ArrayHelper;
     "use strict";
 
     export class ExportGridToExcelService {
         private exportFlag = false;
         static $inject = ["needsWidthFixService"];
+        private columnsToShow: Array<{ columnId: string, index?: number, parentId?: string}> = [];
 
         constructor(private readonly needsWidthFixService: NeedsWidthFix) { }
 
         getExcel(e: IKendoGridExcelExportEvent<any>, _: ILoDashWithMixins, timeout: ng.ITimeoutService, kendoGrid: IKendoGrid<any>) {
-            const columns = e.sender.columns;
+            var columns = e.sender.columns;
 
             if (!this.exportFlag) {
                 e.preventDefault();
-                _.forEach(columns, column => {
-                    if (column.hidden) {
-                        column.tempVisual = true;
-                        e.sender.showColumn(column);
+                _.forEach(columns, (column, i) => {
+                    if (!column.hidden) {
+                        this.columnsToShow.push({ columnId: column.persistId });
+                        return;
                     }
+                    if (column.attributes.parentId === undefined)
+                        return;
+
+                    //look for a parent column and check if any was found, and if any parent column should be visible
+                    var columnsWithMatchingParentId = columns.filter(x => x.persistId === column.attributes.parentId);
+                    if (columnsWithMatchingParentId.length < 1)
+                        return;
+                    if (this.checkIfAllColumnsAreHidden(columnsWithMatchingParentId))
+                        return;
+                    
+                    var index = columns.indexOf(columnsWithMatchingParentId[0]);
```

Comment:
hm... since it is an array, why not reorder all of them?.. In this line you assume that there can be only one dependee, which is true for now but the line above assumes a list of dependees.. might as well make it robust to future changes,